### ISSUE TITLE
Optimize Math.Pow(X, C) in JIT

### ISSUE THREAD

**Issue by EgorBo** [dotnet/coreclr#26434]
I noticed that it's quite popular to use constans with Math.Pow, e.g.
```csharp
double z = Math.Pow(x, 2);
// which is equilent of
double z = x * x;
```
Codegen:
```asm
C.Test1(Double)
    L0000: vzeroupper
    L0003: vmovsd xmm1, qword [rip+0x15]
    L000b: mov rax, 0x7ffec7f7cca0
    L0015: jmp rax

C.Test2(Double)
    L0000: vzeroupper
    L0003: vmulsd xmm0, xmm0, xmm0
    L0007: ret
```

E.g. Burgers benchmark uses it a lot  https://github.com/dotnet/performance/blob/master/src/benchmarks/micro/coreclr/Burgers/Burgers.cs#L22

I am currently converting Math methods in Mono to LLVM intrinsic and noticed it, see https://github.com/mono/mono/pull/16561

**Comment by tannergooding** [comment]
This kind of optimization seems fine to me, provided we are ensuring we remain IEEE compliant and producing a deterministic result.

For example, `Math.Pow(x, 2)` should be fine, However, `Math.Pow(x, 3)` to `x * x * x` would not be.

**Comment by EgorBo** [comment]
@tannergooding yeah, C++ compilers also require "fast math" mode for values other than 2: https://godbolt.org/z/PVr4Nm
